pkg/trafficpolicy/trafficpolicy_test.go

```diff
@@ -437,12 +424,33 @@ func TestMergeInboundPoliciesWithIngress(t *testing.T) {
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			actual := MergeInboundPolicies(true, tc.originalInbound, tc.newIngressInbound...)
-			assert.ElementsMatch(tc.expectedInbound, actual)
+			actual := MergeInboundPolicies(true, tc.originalInbound, tc.newInbound...)
+			assert.True(inboundPoliciesMatch(actual, tc.expectedInbound))
```

Comment:
Could you instead compare the internal slices in the struct directly using `assert.ElementsMatch`. I doubt the existing statement will show the diff in the arrays when the test fails, and would instead show a diff such as `actual: true, expected: false`, which is not very helpful.

Reply:
`assert.ElementsMatch` won't work here I already tried it. The reason being it compares the top level struct ignoring order for it and not for every object in the struct. 

For the unit tests here, the order in which the hostnames appear in the expected to actual list vary ( depending on how they are merged in the `subsetUnion` function and hence the test keeps failing with `ElementsMatch`. I had to come up with my own custom method

Reply:
I am not suggesting to try it for the top level struct, but rather internal slices.

Reply:
I tried that as well and it doesn't work, I can sync with you offline to explain

Reply:
Let me try to see if there is a simple way to resolve this.

Reply:
sure,  let me know what you find. I already tried a few techniques

Reply:
sorted the elements to simplify this